nupm/install.nu

```diff
@@ -23,107 +23,161 @@ def throw-error [
     }
 }
 
-def open-package-file [path: path] {
-    let package_file = $path | path join "package.nuon"
+def open-package-file [dir: path] {
+    let package_file = $dir | path join "package.nuon"
 
     if not ($package_file | path exists) {
-        throw-error $"package_file_not_found(ansi reset):\nno 'package.nuon' found in ($path)"
+        throw-error (
+            [
+                $"package_file_not_found(ansi reset):"
+                $"no 'package.nuon' found in ($dir)"
+            ]
+            | str join (char nl))
     }
 
     let package = open $package_file
 
     log debug "checking package file for missing required keys"
-    let required_keys = [$. $.name $.version $.description $.license $.type]
-    let missing_keys = $required_keys | where {|key| ($package | get -i $key) == null}
+    let required_keys = [$. $.name $.version $.type]
+    let missing_keys = $required_keys
+        | where {|key| ($package | get -i $key) == null}
     if not ($missing_keys | is-empty) {
-        throw-error $"invalid_package_file(ansi reset):\n($package_file) is missing the following required keys: ($missing_keys | str join ', ')"
+        throw-error (
+            [
+                $"invalid_package_file(ansi reset):"
+                ($"($package_file) is missing the following required keys:"
+                    + $" ($missing_keys | str join ', ')")
+            ]
+            | str join)
     }
 
     $package
 }
 
-def prepare-directory [directory: path] {
-    rm --recursive --force $directory
-    mkdir $directory
-}
-
-def copy-directory-to [destination: path] {
-    let source = $in
-
-    log info "copying directory"
-    log debug $"source: ($source)"
-    log debug $"destination: ($destination)"
+# Install list of scripts into a directory
+#
+# Input: Scripts taken from 'package.nuon'
+def install-scripts [
+    pkg_dir: path        # Package directory
+    scripts_dir: path    # Target directory where to install
+    --force(-f)          # Overwrite already installed scripts
+] {
```

Comment:
i would add a type annotation here
```suggestion
]: list<path> -> nothing {
```

Reply:
Currently, it doesn't support installing from pipeline input, but it would be a cool addition. I'll change it to `nothing -> nothing` for now.

Reply:
mm i'm not sure i understand here :thinking: 

the command basically does
```nushell
each {|script|
    let src_path = $pkg_dir | path join $script
    let tgt_path = $scripts_dir | path join $script

    ...
}
```
where each `script` is a `string`, so this means that `install-scripts` takes a `list<string>` as input, right? :open_mouth: 

and when we look at [one of the calls to `install-scripts`](https://github.com/nushell/nupm/pull/18/commits/49c4cf5fc11aeeb028b14c5a2701e90213b8fbfa#diff-030e1728e6809d86850afe7153e0c5d902e00f53645fb6c44dda91da9256b49dR119) as `list<string>`, a.k.a. `$package.scripts` is given to its input :yum: